pygeoapi/openapi.py

```diff
@@ -307,19 +298,183 @@ def get_oas_30(cfg: dict, fail_on_invalid_collection: bool = True) -> dict:
+                'tags': [name],
+                'operationId': f'describe{name.capitalize()}Collection',
+                'parameters': [
+                    {'$ref': '#/components/parameters/f'},
+                    {'$ref': '#/components/parameters/lang'}
+                ],
+                'responses': {
+                    '200': {'$ref': f"{OPENAPI_YAML['oapif-1']}#/components/responses/Collection"},  # noqa
+                    '400': {'$ref': f"{OPENAPI_YAML['oapif-1']}#/components/responses/InvalidParameter"},  # noqa
+                    '404': {'$ref': f"{OPENAPI_YAML['oapif-1']}#/components/responses/NotFound"},  # noqa
+                    '500': {'$ref': f"{OPENAPI_YAML['oapif-1']}#/components/responses/ServerError"}  # noqa
+                }
+            }
+        }
+
+        oas['components']['responses'].update({
+                'Tiles': {
+                    'description': 'Retrieves the tiles description for this collection', # noqa
+                    'content': {
+                        'application/json': {
+                            'schema': {
+                                '$ref': '#/components/schemas/tiles'
+                            }
                         }
                     }
                 }
             }
-        },
-        'parameters': {
+        )
+
+        oas['components']['schemas'].update({
+                'tilematrixsetlink': {
+                    'type': 'object',
+                    'required': ['tileMatrixSet'],
+                    'properties': {
+                        'tileMatrixSet': {
+                            'type': 'string'
+                        },
+                        'tileMatrixSetURI': {
+                            'type': 'string'
+                        }
+                    }
+                },
+                'tiles': {
+                    'type': 'object',
+                    'required': [
+                        'tileMatrixSetLinks',
+                        'links'
+                    ],
+                    'properties': {
+                        'tileMatrixSetLinks': {
+                            'type': 'array',
+                            'items': {
+                                '$ref': '#/components/schemas/tilematrixsetlink' # noqa
+                            }
+                        },
+                        'links': {
+                            'type': 'array',
+                            'items': {'$ref': f"{OPENAPI_YAML['oapit']}#/components/schemas/link"}  # noqa
+                        }
+                    }
+                }
+            }
+        )
+
+    oas['paths'] = paths
+
+    for api_name, api_module in all_apis().items():
+        LOGGER.debug(f'Adding OpenAPI definitions for {api_name}')
+
+        try:
+            sub_oas = api_module.get_oas_30(cfg, locale_)
+            oas['paths'].update(sub_oas['paths'])
+            oas['tags'].extend(sub_oas['tags'])
```

Comment:
How about having each api module's `get_oas_30()` function return a tuple of `list` and `dict` instead? It would make it more explicit than returning a dict with `tags` and `paths` keys inside of it.

So rather than:

`get_oas30() -> dict:`

it would be:

`get_oas30() -> tuple[list[str], dict[str, dict]]`

Reply:
Good idea! Done.